Lib/typing.py

```diff
@@ -935,6 +935,9 @@ def __mro_entries__(self, bases):
         return tuple(res)
 
     def __getattr__(self, attr):
+        if attr in {'__name__', '__qualname__'}:
+            return self._name
```

Comment:
You can't use `_name` in both. `__name__` should be the simple name whereas `__qualname__` should include a `"typing."` prefix, just like Guido requested on the issue.

Reply:
Thanks, fixed.

I am curios about why we should include `typing.` prefix at `__qualname__?

From what I saw `__qualname__`  does not include module name but rather include path to object at scope of module:
```py
class A:
    pass

    class B:
        pass

        class C:
            pass


print(A.__qualname__)
print(A.B.__qualname__)
print(A.B.C.__qualname__)
```
```
A
A.B
A.B.C
```